documentation/debugging.rst

```diff
@@ -0,0 +1,104 @@
+Debugging
+##########
+
+This section is to talk about examples of how you would debug your config.
+
+---------------
+Basic knowledge
+---------------
+
+Sometimes, things aren't as point and click as you might expect it to be.  
+Here are some helpful tips that might help you on your way to making bindings.
+
+
+---------------
+inconsistencies
+---------------
+
+Binder moves down your ``all_includes_file`` sequentially, sometimes you might end up with errors,
+and sometimes you might not.  This is almost always caused by your ``all_includes_file`` being
+different between runs.  The order shouldn't really matter, but nail it down to at least be
+consistent, and then move on to the next step.
+
+--------------
+Build failures
+--------------
+
+Sometimes you might get an error when building your compiled module.  binder has trouble  
+when standard libraries include eachother.  We have already mapped many of these
+interactions in ``types.cc``, but we still might be missing some internal knowledge about
+all of the c++ std library modules out there.  An error like this almost always
+manifests itself with an abundance of errors complaining about ``std::bits`` or ``bits``.
+
+
+So let's go through something I just debugged. Binder seemed to work fine, however the actual
+compiling step failed with the error:
+
+.. code-block:: console
+
+    FAILED: CMakeFiles/statvec.dir/std/complex.o 
+    In file included from std/complex.cpp:1:0:
+    /usr/include/c++/7/bits/stl_construct.h: In function ‘void std::_Destroy(_ForwardIterator, _ForwardIterator)’:
+    **bunch of garbage below here**
+
+1.  Rebuild bindings, but now add the flag ``--annotate-includes`` which will give us much
+    more information than we would normally get
+
+2.  Since we know that ``bits`` is bad for binder, we can just grep for bits in the directory where we
+    generate and compile the bindings.  In the ``examples`` section, this is the same as the directory
+    ``cmake_bindings``.
```

Comment:
Probably a bit of info here would be better approach. I suggest re-wording this along this lines:
... header in bits/ is inner std implementation header that should not be included directly. So if our generated code ends up with such include files it is a sign that Binder `type.cpp` file might need updating by adding additional mapping between std::<some-type> and appropriate std header file.